src/main/java/uk/gov/hmcts/ccd/data/casedetails/query/CaseDetailsQueryBuilderFactory.java

```diff
@@ -1,26 +0,0 @@
-package uk.gov.hmcts.ccd.data.casedetails.query;
-
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
-import uk.gov.hmcts.ccd.data.casedetails.CaseDetailsEntity;
-
-import javax.persistence.EntityManager;
-
-@Component
-public class CaseDetailsQueryBuilderFactory {
-
-    private final UserAuthorisationSecurity userAuthorisationSecurity;
-
-    @Autowired
-    public CaseDetailsQueryBuilderFactory(UserAuthorisationSecurity userAuthorisationSecurity) {
-        this.userAuthorisationSecurity = userAuthorisationSecurity;
-    }
-
-    public CaseDetailsQueryBuilder<CaseDetailsEntity> select(EntityManager em) {
-        return userAuthorisationSecurity.secure(new SelectCaseDetailsQueryBuilder(em));
```

Comment:
@gbenadikar The whole point of this class, with the package-scoped `SelectCaseDetailsQueryBuilder` was that the `secure` step could not programmatically be bypassed outside of this package. By removing that, we now don't have a guaranty that security has been applied. Worse, it's now the responsibility of each repository function to enable the security which makes it optional at best. I don't think this is a change in the right direction, but happy to be proved wrong.

Reply:
@vlaurin thats alright, I have reverted this change now. Please review